src/Tinker.jl

```diff
@@ -475,17 +266,224 @@ function rect_select{T}(c::GtkReactive.Canvas, zr::Signal{ZoomRegion{T}})
     Dict("enabled"=>enabled)
 end
 
+## Functions for handling zoom:
+# Zooms zr to the decimal % entered; view centered around center XY
+function zoom_percent(z::Float64, zr::ZoomRegion, center::XY{Int})
+    # Calculate size of new view
+    fsize = XY(zr.fullview.x.right,zr.fullview.y.right) # full size
+    csize = XY(Int(round(fsize.x/z)), Int(round(fsize.y/z))) # new current size
+    # Calculate center point of new view
+    offset = XY(center.x-Int(round(csize.x/2)),
+                center.y-Int(round(csize.y/2))) # offset of cv
+    # Limit offset
+    if offset.x < 0
+        y = offset.y
+        offset = XY(0, y)
+    elseif offset.x > (fsize.x-csize.x)
+        y = offset.y
+        offset = XY(fsize.x-csize.x, y)
+    end
+    if offset.y < 0
+        x = offset.x
+        offset = XY(x, 0)
+    elseif offset.y > (fsize.y-csize.y)
+        x = offset.x
+        offset = XY(x, fsize.y-csize.y)
+    end
+    
+    return (offset.y+1:offset.y+csize.y, offset.x+1:offset.x+csize.x)
+end # return value can be pushed to a zr
+
+# Sets default center to be the middle of the cv
+function zoom_percent(z::Float64, zr::ZoomRegion)
+    # Calculate cv
+    return zoom_percent(z,zr,find_center(zr))
+end
+
+# Finds rounded center point of the current view of given ZoomRegion
+function find_center(zr::ZoomRegion)
+    range = zr.currentview
+    csize = XY(range.x.right-range.x.left,range.y.right-range.y.left)
+    center = XY(range.x.left+Int(floor(csize.x/2)),
+                range.y.left+Int(floor(csize.y/2)))
+    return center
+end
+
+# Zoom handling
+zpercents = [1.0,1.2,1.5,2.0,2.5,3.0,4.0,8.0]
+global i = 1 # or: const i = Ref(1)
+# Needs to be one i per ctx
+
+function next_zoom{T}(zr::ZoomRegion{T})
+    xzoom = zr.fullview.x.right/(zr.currentview.x.right-
```

Comment:
Why is this just using `fullview.x.right`? I guess because `fullview.x.left` is always (or rather usually, see https://julialang.org/blog/2017/04/offset-arrays) 1? It might be slightly safer to use a real width, e.g., `IntervalSets.width(zr.fullview.x)/IntervalSets.width(zr.currentview.x)`.